src/main/scala/scala/compat/java8/converterImpl/MakesSteppers.scala

```diff
@@ -0,0 +1,160 @@
+trait MakesDoubleSeqStepper extends Any {
+  /** Generates a fresh stepper over `Double`s suitable for sequential use */
+  def stepper: DoubleStepper 
+}
+
+/** Classes or objects implementing this trait create `Double` steppers for map keys suitable for sequential use. */
+trait MakesDoubleKeySeqStepper extends Any {
+  /** Generates a fresh stepper over map keys of type `Double` suitable for sequential use */
+  def keyStepper: DoubleStepper 
+}
+
+/** Classes or objects implementing this trait create `Double` steppers  for map values suitable for sequential use. */
+trait MakesDoubleValueSeqStepper extends Any {
+  /** Generates a fresh stepper over map values of type `Double` suitable for sequential use */
+  def valueStepper: DoubleStepper 
+}
+
+/** Classes or objects implementing this trait create `Int` steppers suitable for sequential use. */
+trait MakesIntSeqStepper extends Any {
+  /** Generates a fresh stepper over `Int`s suitable for sequential use */
+  def stepper: IntStepper 
+}
+
+/** Classes or objects implementing this trait create `Int` steppers for map keys suitable for sequential use. */
+trait MakesIntKeySeqStepper extends Any {
+  /** Generates a fresh stepper over map keys of type `Int` suitable for sequential use */
+  def keyStepper: IntStepper 
+}
+
+/** Classes or objects implementing this trait create `Int` steppers for map values suitable for sequential use. */
+trait MakesIntValueSeqStepper extends Any {
+  /** Generates a fresh stepper over map values of type `Int` suitable for sequential use */
+  def valueStepper: IntStepper 
+}
+
+/** Classes or objects implementing this trait create `Long` steppers suitable for sequential use. */
+trait MakesLongSeqStepper extends Any {
+  /** Generates a fresh stepper over `Long`s suitable for sequential use */
+  def stepper: LongStepper 
+}
+
+/** Classes or objects implementing this trait create `Long` steppers for map keys suitable for sequential use. */
+trait MakesLongKeySeqStepper extends Any {
+  /** Generates a fresh stepper over map keys of type `Long` suitable for sequential use */
+  def keyStepper: LongStepper 
+}
+
+/** Classes or objects implementing this trait create `Long` steppers for map values suitable for sequential use. */
+trait MakesLongValueSeqStepper extends Any {
+  /** Generates a fresh stepper over map values of type `Long` suitable for sequential use */
+  def valueStepper: LongStepper 
+}
+
+/** Classes or objects implementing this trait create generic steppers suitable for sequential or parallel use. */
+trait MakesAnyStepper[A] extends Any {
+  /** Generates a fresh stepper over `A`s that can be efficiently subdivided */
+  def stepper: AnyStepper[A] with EfficientSubstep
+}
+
+/** Classes or objects implementing this trait create generic steppers for map keys suitable for sequential or parallel use. */
+trait MakesAnyKeyStepper[A] extends Any {
```

Comment:
Here's a first pass at simpification: https://github.com/szeiger/scala-java8-compat/commit/daf8bd692e5b2fcd1210ca88ff67446d17de0f94

The separate traits for key / value / generic steppers are still necessary but it now abstracts over the stepper types. The way these typeclasses are used for producing streams, I don't think we can get rid of them easily. If we moved the conversion from steppers to streams down into the typeclasses, it should be possible to remove the special cases for primitive stream converters.